services/blockchain-indexer/methods/dataService/controllers/transactions.js

```diff
@@ -23,21 +23,31 @@ const {
 	Utils: { isEmptyArray, isEmptyObject },
 } = require('lisk-service-framework');
 
-const { getAddressByAny } = require('../../../shared/accountUtils');
+const { confirmAddress } = require('../../../shared/accountUtils');
 
 const dataService = require('../../../shared/dataService');
 
 const getTransactions = async (params) => {
 	try {
-		if (params.senderAddress) {
-			// TODO: Check if this section can be removed
-			const address = await getAddressByAny(params.senderAddress);
-			if (!address) {
-				return {
-					status: NOT_FOUND,
-					data: { error: `Address ${params.senderAddress} not found.` },
-				};
-			}
+		const addressParam = [
+			'senderAddress',
+			'recipientAddress',
+			'address',
+		].filter((item) => typeof params[item] === 'string');
+
+		const addressLookupResult = await Promise.all(
+			addressParam.map(async (param) => {
+				const paramVal = params[param];
+				const address = await confirmAddress(paramVal);
+				return address;
+			}),
+		);
+
+		if (addressLookupResult.includes(false)) {
+			return {
+				status: NOT_FOUND,
+				data: { error: `Account ${params[addressParam[0]]} not found.` },
```

Comment:
Instead of this error message, please send out a `404 NOT FOUND` instead. Technically, if an address clears the regex at the gateway, it can exist on the blockchain. So, this message is incorrect.